### PR TITLE
[#2804] Update headers for requests with Haal Centraal API client

### PR DESCRIPTION
**Pull request by pi-sigma**
We need to pass additional headers for requests made with the Haal Centraal API client (these are not required by the API itself, but needed for integration of other services in connection with the API, e.g. PinkRoccade):

- `x-origin-oin`
- `x-doelbinding`
- `x-verwerking`
- `x-gebruiker`

Taiga: https://taiga.maykinmedia.nl/project/open-inwoner/task/2804

See also the implementation in Open Forms: https://github.com/open-formulieren/open-forms/issues/3731

### CODE REVIEW

**Comment by pi-sigma** [comment]
@alextreme I added a field to the Haal Centraal configuration for the  `x-verwerking` header, as well, in conformity with the Open Forms implementation (https://github.com/open-formulieren/open-forms/issues/3731). Is this useful, or do we know for sure that we don't need this?

**Comment by alextreme** [comment]
@pi-sigma I'm not sure what the purpose of x-verwerking is, so having this working the same as OF seems like a smart idea

File: src/open_inwoner/haalcentraal/api.py [hunk 1]
```diff
@@ -117,6 +117,19 @@ class BRP_2_1(BRPAPI):
 
     def make_request(self, user_bsn: str) -> requests.Response:
         url = "personen"
+
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
```
[the remark this answers is not in the document]
Comment (pi-sigma) [reply src/open_inwoner/haalcentraal/api.py]: You're right about "Content-Type", this is handled by the requests library. The "Accept" header is not, so I'm keeping this.

**Comment by alextreme** [comment]
Okay, it may have been a bit more flexible if GebruikerZelf could also have been made configurable

**Pull request merged**